### DISCUSSION TITLE
Multi form 1 page save all with one click

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2741] [Help] [answered]
If I have 10 forms on-page and I want to write one line for each form when I save the page. Can I do this with a single action button without every form being a subform of the one above it?

right now I can only get all of the forms to save if they are subforms of the form above. this in turn messes up all of my formattings because it tries to put the subform in a new section of the above form.

**Comment by mjashanks** [comment, marked as answer]
This is possible, but a little laborious!

My setup, with video below, shows how 2 rows can be saved to the same table. I only have `Name` and `Address` columns:
1. Create one form
2. Create 4 text fields, withe their `Field` properties set to `Name1`, `Address1`, `Name2`, `Address2`
3. Create a save button, inside the form
4. Click "Define actions"
5. Add a "Save Row" - make sure to choose your form as the data source, and your table
6. Override both columns - `Name = {{ Form.Fields.Name1 }}` and `Address = {{ Form.Fields.Address1 }}`
7. Add another `Save Row` and repeat step 6 for `Name2` 

https://user-images.githubusercontent.com/3524181/134908416-99025cad-be81-41bb-9340-8ba6a06c40f4.mp4

and `Address2`

**Reply by KernOil** [reply]
@mjashanks Thank you this will work for me!
